### PR TITLE
Render /manage using ManageJenkinsAction, handle links using StaplerF…

### PR DESCRIPTION
**Pull request by daniel-beck**
…allback

---

Quoting from #6068: 

> Previously, nothing below "Manage Jenkins" actually had a breadcrumb for it, because `/manage` isn't a ModelObject and the pages below it aren't its children.

and

> I considered having the list of items there, but it not being a `ModelObject` makes it annoying.

This is called [foreshadowing](https://en.wikipedia.org/wiki/Foreshadowing):

> Foreshadowing is a literary device in which a writer gives an advance hint of what is to come later in the story. Foreshadowing often appears at the beginning of a story, or a chapter, and it helps the reader develop expectations about the upcoming events.

This PR uses the [StaplerFallback](https://javadoc.jenkins.io/component/stapler/org/kohsuke/stapler/StaplerFallback.html) feature to ensure all (relative) links from "Manage Jenkins" can be handled. Most links seem to be one of these:

* The `ManagementLink` URL itself. `Jenkins#getDynamic(String)` specifically supports those as an alternative to `RootAction`. This is the basic case.
* Another `RootAction`'s URL (e.g. https://github.com/jenkinsci/script-security-plugin/blob/232f66e9dacfd4e7605e4303ae834c8d83fe8795/src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApprovalLink.java#L45)
* A `PluginImpl`'s `plugin/foo` URL (deprecated)
* A view (e.g. `script.jelly`) of `Jenkins` (expected to exist only in core)

All of these are supported with the `StaplerFallback` solution, at the "cost" of having a new full URL space that can be used to create nonsensical URLs; but Jenkins has many of those already anyway.

A handful of plugins may be incompatible with this approach (by specifying an absolute URL), but the worst that would happen is that there's no breadcrumb inserted. https://github.com/jenkinsci/monitoring-plugin/blob/8c2d48011c54cf05c98b9e5ad925cd7b38604032/src/main/java/org/jvnet/hudson/plugins/monitoring/PluginManagementLink.java#L79-L84 is one such example, but since it renders its own UI, it doesn't matter (and it's actually necessary in this particular case).

Plugins that assume their `ManagementLink` views are rendered at top level _may_ have problems, but except for redirects from `/manage/foo` back to `/foo` (losing the breadcrumb), I'd expect those to be rare edge cases.

### Proposed changelog entries

* Add breadcrumbs to "Manage Jenkins" and children of it. Developers should ensure they use relative links for navigating between pages if they are a child of "Manage Jenkins".

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6126"></a>

### CODE REVIEW

**Comment by timja** [comment]
All looks good in my testing of it. Context menu works, much nicer approach than the other

**Comment by daniel-beck** [comment]
Reopening because there might be renewed interest due to https://github.com/jenkinsci/jenkins/pull/6485#issuecomment-1175251633 / https://github.com/jenkinsci/jenkins/pull/6783#discussion_r914211549.

While this is the nicer implementation, the PR comment in #6068 made an attempt at defining how sidepanels should work. Guidelines like that are necessary to ensure a consistent look and feel across the ecosystem.

**Comment by timja** [comment]
(looks like the spotbugs detector has been disabled by default for the next version but meh it was trivial in this case to fix it, https://github.com/jenkinsci/jenkins/pull/6126/commits/680959bb3d20e2df05dae31c9accc570ab19bd28)

**Comment by timja** [comment]
SystemInfo is missing a breadcrumb

**Review by timja — APPROVED** [review]

**Comment by janfaracik** [comment]
I've given it a try and it's working great! Awesome job guys 👍 

---

Unrelated - Only thing I'm not a fan of relating to the breadcrumb bar is including the current page in it - seems a tad unnecessary.

Screenshot 2022-07-07 at 14 12 37

**Review by NotMyFault — APPROVED** [review]
Looks fine, I guess we don't need to request a security review here :P

**Review by janfaracik — APPROVED** [review]

**Comment by timja** [comment]
@daniel-beck did you want to check the changes I made to your branch?

File: core/src/main/resources/jenkins/model/Jenkins/load-statistics.jelly
Comment (daniel-beck) [inline core/src/main/resources/jenkins/model/Jenkins/load-statistics.jelly:29]: Why `href`? Jellydoc says it can be omitted.

File: core/src/main/resources/jenkins/model/Jenkins/systemInfo.jelly
Comment (daniel-beck) [inline core/src/main/resources/jenkins/model/Jenkins/systemInfo.jelly:32]: Why `href`? Jellydoc says it can be omitted.

**Comment by daniel-beck** [comment]
> Personally it feels better to me having the current page in there, more balanced even if it's not really useful =/

Agreed, from the attempt to establish guidelines in #6068:

> Every page reachable from "Manage Jenkins" has a "self" breadcrumb. Whether that is realistically doable manually, or whether I just rewrite a bunch of stuff to `RootAction` is TBD. (Right now it looks weird when it's a simple view with parent and self breadcrumbs, still need to insert a divider)

In particular, you will get a breadcrumb for a `ModelObject`'s `index` view automatically. This is why _Configure Global Tools_ has it, and _Configure System_ does not.

File: core/src/main/resources/jenkins/model/Jenkins/systemInfo.jelly
Comment (timja) [reply core/src/main/resources/jenkins/model/Jenkins/systemInfo.jelly:32]:
```suggestion
    <l:breadcrumb title="${%System Information}"/>
```

File: core/src/main/resources/lib/hudson/scriptConsole.jelly
Comment (timja) [reply core/src/main/resources/lib/hudson/scriptConsole.jelly:33]:
```suggestion
    <l:breadcrumb title="${%Script Console}"/>
```

File: core/src/main/resources/jenkins/model/Jenkins/load-statistics.jelly
Comment (timja) [reply core/src/main/resources/jenkins/model/Jenkins/load-statistics.jelly:29]:
```suggestion
    <l:breadcrumb title="${%Load Statistics}"/>
```

Comment (timja) [reply core/src/main/resources/jenkins/model/Jenkins/load-statistics.jelly]: it was to make it a link, not that it was very useful, I've removed for now

**Comment by timja** [comment]
Look good to you @daniel-beck ?

**Comment by NotMyFault** [comment]
This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

File: core/src/main/java/hudson/model/ManageJenkinsAction.java
Comment (jglick) [inline core/src/main/java/hudson/model/ManageJenkinsAction.java:68]: This has been causing some confusion, because now there are two URLs to access every admin-type page, with and without the `manage/` infix, and various pages link to one or the other inconsistently.

Comment (daniel-beck) [reply core/src/main/java/hudson/model/ManageJenkinsAction.java:68]: No good, compatible way around this, unfortunately.

Comment (jglick) [reply core/src/main/java/hudson/model/ManageJenkinsAction.java:68]: Can we not serve 302s to the `/manage/*` versions?

**Pull request merged**
